synapse/handlers/admin.py

```diff
@@ -247,6 +248,21 @@ async def export_user_data(self, user_id: str, writer: "ExfiltrationWriter") ->
                 )
                 writer.write_state(room_id, event_id, state)
 
+        # Get the user profile
+        profile = await self.get_user(UserID.from_string(user_id))
+        assert profile
+        writer.write_profile(profile)
+
+        # Get all devices the user has
+        devices = await self._device_handler.get_devices_by_user(user_id)
+        writer.write_devices(devices)
+
+        # Get all connections the user has
+        connections = await self.get_whois(UserID.from_string(user_id))
+        writer.write_connections(
+            connections["devices"][""]["sessions"][0]["connections"]
+        )
```

Comment:
Would calling this "sessions" be clearer?

Reply:
Im not sure if session is the right word.
IMO are devices (acces_toke) sessions. But connection are additionally recorded and stored in the database.

Reply:
I'm not really sure of the right word either. 😢 I unfortunately don't have a great solution. The spec around this data seems to imply it should be tied back to devices, but it doesn't seem that Synapse returns it that way?

Reply:
I also have read the [spec](https://spec.matrix.org/v1.5/client-server-api/#get_matrixclientv3adminwhoisuserid) and was irritated. Synapse returns all last connections without devices. The spec refers to devices and their last connections.

The spec has devices (access_token) -> sessions -> connections. It is also unclear to me how a device can have multiple sessions. Connections is clear for me.